### PR TITLE
Added single quote and 'rem' style comments.

### PR DESCRIPTION
**Pull request by jimmckay**
Fixed parsing with multiple Note sections.
Added .xojo_image, .rbbas, .rbfrm, .rbmnu, .rbres, .rbtbar file extensions (to include old realbasic file names)
Updated demo and test files with added comment styles.
Also added WeakAddressOf operator keyword.

### CODE REVIEW

File: lib/rouge/lexers/xojo.rb [hunk 1]
```diff
@@ -8,7 +8,7 @@ class Xojo < RegexLexer
       desc "Xojo"
       tag 'xojo'
       aliases 'realbasic'
-      filenames '*.xojo_code', '*.xojo_window', '*.xojo_toolbar', '*.xojo_menu'
+      filenames '*.xojo_code', '*.xojo_window', '*.xojo_toolbar', '*.xojo_menu', '*.xojo_image', '*.rbbas', '*.rbfrm', '*.rbmnu', '*rbres', '*rbtbar'
```
Comment (pyrmont) [inline lib/rouge/lexers/xojo.rb]: Are those last two supposed to have a dot after the asterisk?

File: lib/rouge/demos/xojo [hunk 2]
```diff
@@ -1,7 +1,8 @@
 Dim f As FolderItem
 f = GetOpenFolderItem(FileTypes1.jpeg) // defined in the File Type Set editor
+rem - we should check for nil!
```
Comment (pyrmont) [inline lib/rouge/demos/xojo]: What's the significance of the hyphen?

File: lib/rouge/lexers/xojo.rb (hunk 1 above)
Comment (jimmckay) [reply lib/rouge/lexers/xojo.rb]: Yes. I've fixed and pushed the change.

File: lib/rouge/demos/xojo (hunk 2 above)
Comment (jimmckay) [reply lib/rouge/demos/xojo]: No significance. That line should be considered a "rem comment", so any text is allowed.

**Comment by pyrmont** [comment]
Thanks for improving Rouge, @jimmckay :)

**Pull request merged**